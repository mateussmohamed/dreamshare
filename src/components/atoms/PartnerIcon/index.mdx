---
name: PartnerIcon
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import PartnerIcon from './index'

# PartnerIcon
<PropsTable of={PartnerIcon} />

<Playground>
  <PartnerIcon />
</Playground>
