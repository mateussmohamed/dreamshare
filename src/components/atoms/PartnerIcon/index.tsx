import styled from 'styled-components'

interface PartnerIconProps {
  icon: string;
}

export const PartnerIcon =
  styled.div <
  PartnerIconProps >
  `
  width: 32px;
  height: 32px;
  border-radius: 32px;
  background-color: #f35e5a;
  border: 4px solid #ffffff;
  position: absolute;
  right: -1px;
  bottom: -1px;
  background-image: url(${props => props.icon});
  background-size: 16px 16px;
  background-position: center center;
  background-repeat: no-repeat;
`
