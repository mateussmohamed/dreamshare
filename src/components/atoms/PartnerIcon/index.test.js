import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import PartnerIcon from './index'

describe('[<PartnerIcon />]', () => {
  const wrap = (props = {}) => shallow(<PartnerIcon {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({})

    expect(wrapper).toMatchSnapshot()
  })
})
