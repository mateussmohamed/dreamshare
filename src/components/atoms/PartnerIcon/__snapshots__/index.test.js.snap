// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<PartnerIcon />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "PartnerIcon-sc-1uwsdkq-0",
        "isStatic": false,
        "rules": Array [
          "width:32px;height:32px;border-radius:32px;background-color:#f35e5a;border:4px solid #ffffff;position:absolute;right:-1px;bottom:-1px;background-image:url(",
          [Function],
          ");background-size:16px 16px;background-position:center center;background-repeat:no-repeat;",
        ],
      },
      "displayName": "PartnerIcon",
      "render": [Function],
      "styledComponentId": "PartnerIcon-sc-1uwsdkq-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
